proving_system/stark/src/prover.rs

```diff
@@ -158,11 +157,28 @@ where
         .map(|fri_commitment| fri_commitment.merkle_tree.root.clone())
         .collect();
 
+    let q_0 = transcript_to_usize(transcript) % 2_usize.pow(lde_root_order);
+    transcript.append(&q_0.to_be_bytes());
+
+    let deep_consistency_check = build_deep_consistency_check(
+        q_0,
+        &lde_roots_of_unity_coset,
+        &lde_trace,
+        &composition_poly_even,
+        &composition_poly_odd,
+    );
+
     let query_list = (0..air.context().options.fri_number_of_queries)
-        .map(|_| {
-            // * Sample q_1, ..., q_m using Fiat-Shamir
-            let q_i = transcript_to_usize(transcript) % 2_usize.pow(lde_root_order);
-            transcript.append(&q_i.to_be_bytes());
+        .enumerate()
+        .map(|(i, _)| {
```

Comment:
I think te `enumerate()` can be removed and use the index directly from the `(0..air.context().options.fri_number_of_queries)` range